From 81b0f2e30dc65b496966933e74156ee83491e4e8 Mon Sep 17 00:00:00 2001
From: "penikelapati.kumar" <penikelapati.kumar@lge.com>
Date: Tue, 1 Jun 2021 16:00:56 +0530
Subject: [PATCH] Fix connman crash when interface removed

:Release Notes:
Fix connman crash when interface path length is zero

:Detailed Notes:
Connman crasing when interface path lenght is zero.

:Testing Performed:
Build successfully.
Ethrenet/wifi connection successfully.

:QA Notes:
N/A

:Issues Addressed:
[PLAT-144536] [RDX_CRASH][webos] /usr/sbin/connmand

Upstream-Status: Inappropriate [webOS specific]
---
 gsupplicant/supplicant.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gsupplicant/supplicant.c b/gsupplicant/supplicant.c
index 59b02c2..27e9e55 100644
--- a/gsupplicant/supplicant.c
+++ b/gsupplicant/supplicant.c
@@ -7028,7 +7028,7 @@ int g_supplicant_interface_remove(GSupplicantInterface *interface,
 	struct interface_data *data;
 	int ret;
 
-	if (!interface)
+	if (!interface || !(interface->path) || (strlen(interface->path) == 0))
 		return -EINVAL;
 
 	if (!system_available)
-- 
2.26.0

